src/test/java/de/zalando/aruha/nakadi/repository/kafka/KafkaRepositoryTest.java

```diff
@@ -0,0 +1,227 @@
+package de.zalando.aruha.nakadi.repository.kafka;
+
+import de.zalando.aruha.nakadi.NakadiException;
+import de.zalando.aruha.nakadi.domain.Cursor;
+import de.zalando.aruha.nakadi.domain.Topic;
+import de.zalando.aruha.nakadi.domain.TopicPartition;
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+import kafka.javaapi.OffsetRequest;
+import kafka.javaapi.consumer.SimpleConsumer;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.api.GetChildrenBuilder;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.PartitionInfo;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Future;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class KafkaRepositoryTest {
+
+    public static final String MY_TOPIC = "my-topic";
+    public static final String ANOTHER_TOPIC = "another-topic";
+
+    @SuppressWarnings("unchecked")
+    public static final ProducerRecord EXPECTED_PRODUCER_RECORD = new ProducerRecord(MY_TOPIC, 0, "0", "payload");
+
+    private static final Set<PartitionState> PARTITIONS;
+
+    static {
+        PARTITIONS = new HashSet<>();
+
+        PARTITIONS.add(new PartitionState(MY_TOPIC, 0, 40, 42));
+        PARTITIONS.add(new PartitionState(MY_TOPIC, 1, 100, 200));
+
+        PARTITIONS.add(new PartitionState(ANOTHER_TOPIC, 1, 0, 100));
+        PARTITIONS.add(new PartitionState(ANOTHER_TOPIC, 5, 12, 60));
+        PARTITIONS.add(new PartitionState(ANOTHER_TOPIC, 9, 99, 222));
+    }
+
+    public static final List<Cursor> MY_TOPIC_VALID_CURSORS = asList(cursor("0", "41"), cursor("1", "100"), cursor("1", "200"), cursor("1", "101"));
+    public static final List<Cursor> ANOTHER_TOPIC_CURSORS = asList(cursor("1", "0"), cursor("1", "100"), cursor("5", "12"), cursor("9", "100"));
+    public static final List<Cursor> MY_TOPIC_INVALID_CURSORS = asList(cursor("0", "39"), cursor("1", "0"), cursor("1", "99"), cursor("1", "201"));
+
+    public static final long LATEST_TIME = kafka.api.OffsetRequest.LatestTime();
+    public static final long EARLIEST_TIME = kafka.api.OffsetRequest.EarliestTime();
+
+
+    private final KafkaRepository kafkaRepository;
+    private final KafkaProducer kafkaProducer;
+
+    public KafkaRepositoryTest() {
```

Comment:
1) It would be difficult to create unit test for that, as it calls static methods so we will need to bring PowerMock magic
2) We do have "integration" acceptance test for that :)